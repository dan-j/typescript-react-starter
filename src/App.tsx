import React from 'react';
import CSSModules from 'react-css-modules';
const styles = require('./_App.scss');

interface AppProps extends CSSModules.InjectedCSSModuleProps {
    name?: String;
}

const App: React.StatelessComponent<AppProps> = ({ name }) => {

    let greg;
    if (name === 'Greg') {
        greg = <h3>Your name is Greg!</h3>;
    }

    return (
        <div>
            <p styleName="greeting">Hello, {name}</p>
            {greg}
        </div>
    );
};

App.defaultProps = {
    name: 'World',
};

export default CSSModules(App, styles);
