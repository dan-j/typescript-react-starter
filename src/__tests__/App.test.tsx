import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json/';

import App from '../App';

describe('<App />', () => {
    it('Renders a default greeting', () => {
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Renders a personalised greeting', () => {
        const wrapper = shallow(<App name="Dan" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

