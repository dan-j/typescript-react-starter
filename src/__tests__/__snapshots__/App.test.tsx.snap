// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders a default greeting 1`] = `
<div>
  <p
    className="greeting"
  >
    Hello, 
    World
  </p>
</div>
`;

exports[`<App /> Renders a personalised greeting 1`] = `
<div>
  <p
    className="greeting"
  >
    Hello, 
    Dan
  </p>
</div>
`;
